ambry-rest/src/test/java/com.github.ambry.rest/NettyResponseChannelTest.java

```diff
@@ -409,42 +409,47 @@ public void setStatusTest() {
   public void errorResponseTest() {
     EmbeddedChannel channel = createEmbeddedChannel();
     for (RestServiceErrorCode errorCode : RestServiceErrorCode.values()) {
-      HttpHeaders httpHeaders = new DefaultHttpHeaders();
-      httpHeaders.set(MockNettyMessageProcessor.REST_SERVICE_ERROR_CODE_HEADER_NAME, errorCode);
-      channel.writeInbound(
-          RestTestUtils.createRequest(HttpMethod.HEAD, TestingUri.OnResponseCompleteWithRestException.toString(),
-              httpHeaders));
-      HttpResponse response = channel.readOutbound();
-      HttpResponseStatus expectedStatus = getExpectedHttpResponseStatus(errorCode);
-      assertEquals("Unexpected response status", expectedStatus, response.status());
-      boolean containsFailureReasonHeader = response.headers().contains(NettyResponseChannel.FAILURE_REASON_HEADER);
-      if (expectedStatus == HttpResponseStatus.BAD_REQUEST) {
-        assertTrue("Could not find failure reason header.", containsFailureReasonHeader);
-      } else {
-        assertFalse("Should not have found failure reason header.", containsFailureReasonHeader);
-      }
-      if (HttpStatusClass.CLIENT_ERROR.contains(response.status().code())) {
-        assertEquals("Wrong error code", errorCode,
-            RestServiceErrorCode.valueOf(response.headers().get(NettyResponseChannel.ERROR_CODE_HEADER)));
-      } else {
-        assertFalse("Should not have found error code header",
-            response.headers().contains(NettyResponseChannel.ERROR_CODE_HEADER));
-      }
-      if (response instanceof FullHttpResponse) {
-        // assert that there is no content
-        assertEquals("The response should not contain content", 0,
-            ((FullHttpResponse) response).content().readableBytes());
-      } else {
-        HttpContent content = channel.readOutbound();
-        assertTrue("End marker should be received", content instanceof LastHttpContent);
-      }
-      assertNull("There should be no more data in the channel", channel.readOutbound());
-      boolean shouldBeAlive = !NettyResponseChannel.CLOSE_CONNECTION_ERROR_STATUSES.contains(expectedStatus);
-      assertEquals("Channel state (open/close) not as expected", shouldBeAlive, channel.isActive());
-      assertEquals("Connection header should be consistent with channel state", shouldBeAlive,
-          HttpUtil.isKeepAlive(response));
-      if (!shouldBeAlive) {
-        channel = createEmbeddedChannel();
+      for (boolean includeExceptionMessageInResponse : new boolean[]{true, false}) {
```

Comment:
most of the changes in this method are just an extra level of indentation, except for line 412, 415-417, and 425.